index.d.ts

```diff
@@ -0,0 +1,208 @@
+/// <reference types="node" />
+/// <reference types="aws-lambda" />
+/// <reference types="connect" />
+
+import { IncomingMessage, ServerResponse } from 'http';
+import { Handler } from 'aws-lambda';
+import { ErrorHandleFunction } from 'connect';
+
+export default agent;
+
+declare const agent: Agent;
+
+declare class Agent implements Taggable, StartSpanFn {
+  // Configuration
+  start (options?: AgentConfigOptions): Agent;
+  isStarted (): boolean;
+
+  // Data collection hooks
+  middleware: { connect (): ErrorHandleFunction };
+  lambda (handler: Handler): Handler;
+  lambda (type: string, handler: Handler): Handler;
+  handleUncaughtExceptions (
+    fn: (err: Error) => void
+  ): void;
+
+  // Errors
+  captureError (
+    err: Error | string | ParameterizedMessageObject,
+    callback?: CaptureErrorCallback
+  ): void;
+  captureError (
+    err: Error | string | ParameterizedMessageObject,
+    options?: CaptureErrorOptions,
+    callback?: CaptureErrorCallback
+  ): void;
+
+  // Transactions
+  startTransaction (
+    name?: string,
+    type?: string,
+    options?: TransactionOptions
+  ): Transaction | null;
+  setTransactionName (name: string): void;
+  endTransaction (result?: string, endTime?: number): void; // TODO: Should we allow number as well for result?
```

Comment:
I think it makes sense to align these types with whatever apm-server accepts. 🤔

Reply:
The APM Server only supports string, but we're stringifying the result before sending it to the server, and I could see a use-case for using an exit or status code as a result. Allowing people to parse in a number (that we stringify for them internally) might be a normal use-case?

Reply:
Gotcha. Sounds reasonable to me.